### DISCUSSION TITLE
Migrating nginxproxymanager LXC to new install

### DISCUSSION THREAD

**Discussion by bas89** [tteck/Proxmox#3028] [Q&A] [answered]
I'm having serious updating issues that seem to be only happen on my install (error on line 143 when building frontend when updating 2.11.1 to 2.11.2 of npm LXC).

I rather would like to start with a new install and migrate my old installation. However, copying /data/ and /etc/letsencrypt/ folders doesn't seem to suffice, as the web interface as well as the nginx server don't start. Any ideas what is missing? I'm having a SQlite DB in /data/, so that shouldn't be the problem.

**Comment by Mellowlynx** [comment, marked as answer]
Are both npm versions the same?
A view weeks ago, I migrated one for a friend using [this guide](https://github.com/NginxProxyManager/nginx-proxy-manager/discussions/1529) and that worked fine.

I did, however, stop the `npm.service` before packing up and restoring both folders to be sure.

**Reply by bas89** [reply]
The npm version of the LXC with the update issue (2.11.1) is indeed different from the new one (2.11.2) -- but can I force to freshly install a specific version? Thanks for the tip regarding to `systemctl stop npm.service` first.

I tried to edit tteck's `nginxproxymanager.sh`, setting the appropriate line to `RELEASE="2.11.1"`, but it still installs the latest version 2.11.2.

**Reply by Mellowlynx** [reply]
As far as I know, there is no way to install an older/specific version through the script.
This is a good thing in my eyes, but for debugging it's not so good.

@tteck is there a way to select a specific version?

**Reply by tteck** [reply]
No, you have the option to install v2.10.4 or the latest.

**Reply by Mellowlynx** [reply]
In that case, I would add everything fresh, bas89. Not fun (I know the pain)
You could keep trying and changing the _wget url_ for example, but a fresh start might end up faster, depending on your host count.

**Reply by bas89** [reply]
Thanks a lot guys. Now the following approach worked out:

1. Backup old folders. In pve shell, `mount /dev/mapper/pve-vm--XXX--disk--0 /media`, `cd /media`, `tar czvf data.tar.gz data/`, `cd etc`, `tar czvf letsencrypt.tar.gz letsencrypt` and move the archives to `/` on pve. `umount /media`.
2. Delete the old LXC, created a new one with tteck's script from scratch with latest version (2.11.2, which is just a minor update compared to my old LXC). I set the MAC address the same as the old LXC. Shut it down.
3. Mount the new LXC's disk as above. Back-up the initial folders with `cd /media`, `mv data data_bak`, `cd ../etc`, same for letsencrypt. This ensures you don't have some files from the initial install there as you want to replace the whole dir (instead of just replacing all the files in there).
4. Unzip the folders like `cd /media`, `tar xvf data.tar.gz` and `cd` to the `etc` folder for letsencrypt and do the same. Finally, `umount /media`. 
5. Boot the new LXC again. 

For me, it worked although there was a minor version difference.

**Reply by Mellowlynx** [reply]
Good to read that, happy you fixed it.
